assets/javascripts/modules/autoComplete.js

```diff
@@ -114,6 +114,7 @@ var closeSuggestionsAndFocus = function () {
  */
 var clearInput = function () {
   $autoCompleteInputElem.val('').focus();
+  $autoCompleteInputElem.data('suggestionValue', '');
```

Comment:
do you want the data to be an empty string? `$autoCompleteInputElem.removeData('suggestionValue');` would remove the data rather then set to an empty value.